lib/internal/streams/transform.js

```diff
@@ -244,3 +246,23 @@ Transform.prototype._read = function() {
     callback();
   }
 };
+
+Transform.by = function by(asyncGeneratorFn, options) {
+  let _resolve;
+  let _promise = new Promise((resolve) => _resolve = resolve);
+  return from(Duplex, asyncGeneratorFn(async function*() {
+    while (true) {
+      const { chunk, done, cb } = await _promise;
+      if (done) return process.nextTick(cb);
```

Comment:
Can we avoid repeating process.nextTick maybe?
```suggestion
      process.nextTick(cb);
      if (done) return;
```